pkg/contracts/src/OsmoticController.sol

```diff
@@ -107,146 +79,63 @@ contract OsmoticController is Initializable, OwnableUpgradeable, PausableUpgrade
-
-    function lockBalance(address _token, uint256 _amount) public whenNotPaused {
-        IStaking staking = stakingFactory.getOrCreateInstance(_token);
-
-        _lockBalance(staking, msg.sender, _amount);
-    }
-
-    function unlockBalance(address _token) public whenNotPaused {
-        require(participantSupportedPools[msg.sender] == 0, "OsmoticController: cannot unlock while supporting a pool");
-
-        (uint256 lockedBalance,) = getStakingLock(_token, msg.sender);
-
-        _unlockBalance(stakingFactory.getInstance(_token), msg.sender, lockedBalance);
-    }
-
-    /* *************************************************************************************************************************************/
-    /* ** Locking and Supporting Functions                                                                                               ***/
-    /* *************************************************************************************************************************************/
-
-    function lockAndSupport(
-        OsmoticPool _pool,
-        uint256 _lockedAmount,
-        ParticipantSupportUpdate[] calldata _participantUpdates
-    ) external whenNotPaused {
-        lockBalance(address(_pool.governanceToken()), _lockedAmount);
-
-        _pool.updateProjectSupports(_participantUpdates);
-    }
-
-    function unsupportAndUnlock(OsmoticPool _pool, ParticipantSupportUpdate[] calldata _participantUpdates)
+    function createMimeToken(string calldata name, string calldata symbol, bytes32 merkleRoot)
         external
         whenNotPaused
+        returns (address)
     {
-        _pool.updateProjectSupports(_participantUpdates);
-
-        unlockBalance(address(_pool.governanceToken()));
-    }
-
-    /* *************************************************************************************************************************************/
-    /* ** Participant Supported Pools Functions                                                                                          ***/
-    /* *************************************************************************************************************************************/
-
-    function increaseParticipantSupportedPools(address _participant) external whenNotPaused onlyPool(msg.sender) {
-        participantSupportedPools[_participant]++;
-
-        emit ParticipantSupportedPoolsChanged(_participant, participantSupportedPools[_participant]);
-    }
+        MimeToken token = MimeTokenFactory(mimeTokenFactory).createMimeToken(name, symbol, merkleRoot);
+        token.transferOwnership(msg.sender);
+
+        return address(token);
+    }
+
+    function createPool(
+        string calldata name,
+        string calldata symbol,
+        bytes32 merkleRoot,
+        address fundingToken,
+        bool isPrivate,
+        OsmoticParams calldata params
```

Comment:
Maybe it would be better to pass the encoded initialized params as we were doing before. 

This way if we update the initialize params of either the osmotic pool / mime token for some reason  in the future we don't have to upgrade the controller to support those new parameters

Reply:
I think it make sense to use the encoded params instead. But the mime token is one of the params of the osmotic pool so I'm not sure if we can handle the encoding/decoding of the initialize function in a generic way. Maybe we should remove this method and keep the three creation function as separate steps (token, list, pool).